02-deploying-an-app/004-use-circleci-to-upgrade-app.md

```diff
@@ -0,0 +1,109 @@
+
+A Service Account is created in the [namespace creation github repository](https://github.com/ministryofjustice/cloud-platform-environments/tree/master/namespaces).
+```bash
+  $ kubectl get sa --namespace $ns
+  NAME       SECRETS   AGE
+  circleci   1         3h
+  $ kubectl get sa/circleci --namespace reference-app -o yaml
+  apiVersion: v1
+  kind: ServiceAccount
+  ...
+  secrets:
+  - name: circleci-token-prlkp
+  $ kubectl get secrets --namespace $ns
+  NAME                   TYPE                                  DATA      AGE
+  circleci-token-prlkp   kubernetes.io/service-account-token   3         3h
+  $ kubectl get secrets/circleci-token-prlkp --namespace $ns -o yaml
+  ...
+  namespace: cm..cA==
+  token: ZX...EE=
+```
+
+### Migration to CircleCI 2.0
+We are using CircleCI 2.0 if you already have a CircleCI circle.yml please [migrate](https://circleci.com/docs/2.0/migration/) your project to 2.0.
+
+### Linking Repository to CircleCI
+MoJ has as an account with CircleCI, please login to [CircleCI](https://circleci.com/dashboard) using GitHub credentials. Select project, and if config.yml is in the repo CircleCI will build and run tests.
+
+### Adding the config.yml
+CircleCI uses a YAML file to identify how you want your testing environment set up and what tests you want to run. On CircleCI 2.0, this file must be called config.yml and must be in a hidden folder called .circleci .
+
+### Add variables to CircleCI
+From Builds click the cog and select Enviroment Variables under Build Settings. The variables needed to add are.
+- AWS_DEFAULT_REGION
+- AWS_ACCESS_KEY_ID
+- AWS_SECRET_ACCESS_KEY
+- ECR_ENDPOINT
+- CA_CERT
+- CLUSTER_NAME
+- TOKEN
+
+### Creating the config.yml
+[Tutorial](https://circleci.com/docs/2.0/tutorials/) on creating a config.yml file. The cloud platform reference app, has the following steps. As long as you are build a docker image you can configure circle however you wish. The only extra code you will need to add are the Upload to ECR and Deploy to Kubernetes.
+- Create the remote docker environment
+- Build environment and install requirements
+- Build image and cache
+- Run tests
+- Upload to ECR
+```bash
+deploy:
+    name: Push application Docker image
+    environment:
+    command: |
+    login="$(aws ecr get-login)"
+    ${login}
+
+    docker tag app "${ECR_ENDPOINT}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_SHA1}"
+    docker push "${ECR_ENDPOINT}/${CIRCLE_PROJECT_REPONAME}:${CIRCLE_SHA1}"
+
+    if [ "${CIRCLE_BRANCH}" == "master" ]; then
+        docker tag app "${ECR_ENDPOINT}/${CIRCLE_PROJECT_REPONAME}:latest"
+        docker push "${ECR_ENDPOINT}/${CIRCLE_PROJECT_REPONAME}:latest"
+    fi
+```
```

Comment:
Think there's a new naming convention for tagging docker images. https://github.com/ministryofjustice/cloud-platform-reference-app/blob/master/.circleci/config.yml#L45